consensus/src/round_manager.rs

```diff
@@ -485,22 +487,42 @@ impl RoundManager {
                     .remote_peer(author),
                 "Local state {}, remote state {}", local_sync_info, sync_info
             );
-            // Some information in SyncInfo is ahead of what we have locally.
-            // First verify the SyncInfo (didn't verify it in the yet).
-            sync_info
-                .verify(&self.epoch_state().verifier)
-                .map_err(|e| {
-                    error!(
-                        SecurityEvent::InvalidSyncInfoMsg,
-                        sync_info = sync_info,
-                        remote_peer = author,
-                        error = ?e,
-                    );
-                    VerifyError::from(e)
-                })?;
+
+            let highest_commit_round = sync_info.highest_commit_round();
+            let mut sync_commit_certs = false;
```

Comment:
@vusirikala - The updated logic looks convoluted and hard to follow. I am wondering if its cleaner if we split up the `sync_up` function into two functions `sync_up_non_commit_cert` and `sync_up_commit_cert`. One thing to take care of is avoid double verification in this and also, we probably need to split the `add_certs` function accordingly. Thoughts? cc - @zekun000